docs/managedidentity_public_api.md

```diff
@@ -0,0 +1,206 @@
+
+# Managed Identity Public API Design Specification
+
+The purpose of this file is to go over the changes required for adding the Managed Identity feature to MSAL GO
+
+## Public API
+
+The public API will be quite small. Based on the Java and .NET implementations, there is only 1 exposed method, **acquireTokenForManagedIdentity()**
+
+```go
+// Acquires tokens from the configured managed identity on an azure resource.
+//
+// Resource: scopes application is requesting access to
+// Options: [WithClaims]
+func (client Client) AcquireToken(context context.Context, resource string, options ...AcquireTokenOption) (base.AuthResult, error) {
+    return base.AuthResult{}, nil
+}
+
+// Source represents the managed identity sources supported.
```

Comment:
Pls move this to a separate section (maybe an appendix), as this API is really only meant for Azure SDK and for testing purposes.

Reply:
Not sure I follow, should the public exposed data not have some documentation provided with it?